src/web_peer_connection.cpp

```diff
@@ -666,6 +666,7 @@ void web_peer_connection::handle_redirect(int const bytes_left)
 		file_index_t const file_index = m_file_requests.front().file_index;
 
 		location = aux::resolve_redirect_location(m_url, location);
+		location = maybe_url_encode(location);
```

Comment:
I think there should be a comment here explaining that this is a work-around for buggy HTTP servers, that fail to encode the redirect URL.